src/snapshot.js

```diff
@@ -69,6 +69,23 @@ export function getPlayerSnapshot(player) {
   }
   snapshotObject.suppressedTracks = suppressedTracks;
 
+  // iOS Safari will change caption mode to 'showing' if a user previously
+  // turned captions on manually for that video source, so this will re-disable
+  // them in case that occurs during ad playback
+  if (videojs.browser.IS_IOS && !Array.isArray(tracks)) {
+    tracks.one('change', function(event) {
+      if (player.ads.state === 'ad-playback') {
+        const trackList = event.target.tracks_;
```

Comment:
I used the `tracks_` property here because I needed an array to loop through with `forEach()`. Is there another way to access the array of TextTracks without using that private property?